srcpkgs/base-files/files/vkpurge

```diff
@@ -13,97 +13,120 @@ esac
 usage()
 {
 	cat <<_EOF
-Usage: $progname <target> [<version>]
+Usage: $progname <MODE> [<VERSION>]
 
-Targets:
- list			Lists old installed kernels.
- rm <version...|all>	Remove kernel <version> or all old kernels.
+MODE
+ list [<VERSION>]		lists old kernels
+ rm (all | <VERSION>...)	remove old kernels
 
-Example:
+Examples:
 	$ $progname list
-	$ $progname rm 4.9.81_1
+	$ $progname list 4.18.*
+	$ $progname rm all
+	$ $progname rm 4.18.17_1 4.18.17_2
+	$ $progname rm 4.18.17_*
+	$ $progname rm 4.18.*
 _EOF
 	exit 1
 }
 
 list_kernels()
 {
-	local k kpkg installed kver _f
-
-	for k in /boot/$KIMAGE-*; do
-		_f=$(basename $k)
-		kver=$(echo ${_f}|sed -e 's|-rc|rc|')
-		installed=$(xbps-query -o "$k" 2>/dev/null)
-		[ -n "$installed" ] && continue
-		kpkg="$(xbps-uhelper getpkgversion ${_f} 2>/dev/null)"
+	local installed
+	local kpkg
+	local k
+
+	if [ "$1" = "all" ]; then
+		shift
+	fi
+
+	if [ -n "$1" ] && ! echo "$1" | grep -Eq '^([0-9]+\.+){1,2}(\*|[0-9]+_([0-9]+|\*))$'; then
+		echo "$1"
+		exit 1
+	fi
+
+	installed="$(xbps-query -o "/boot/$KIMAGE-*" 2>/dev/null | awk '{print $2}')"
+
+	for k in /boot/"$KIMAGE"-*; do
+		(echo "$installed" | grep -q "$k") && continue
```

Comment:
Needless subshell.

Reply:
You might consider `grep -F` here and later since the pattern should be treated as a fixed string not a regex.